torch/csrc/jit/passes/batch_mm.cpp

```diff
@@ -201,6 +201,48 @@ struct TreeToken {
     }
     return matmuls;
   }
+
+  // Checks whether a tree merge is safe by trying to relocate all nodes in the
+  // tree together.
+  bool canMergeTree(const AliasDb& db) {
+    std::vector<Node*> queue{node};
+    while (!queue.empty()) {
+      auto n = queue.back();
+      queue.pop_back();
+      if (n->matches("aten::t(Tensor self) -> Tensor")) {
+        Node* input_node = n->input()->node();
+        JIT_ASSERT(input_node->matches(
+            "aten::mm(Tensor self, Tensor mat2) -> Tensor"));
+        if (!n->moveAfterTopologicallyValid(input_node, db)) {
```

Comment:
Can we please use `couldMove...` instead? Just so that we don't do stupid transformations to the trace only to realize we can't do them all halfway through.